### ISSUE TITLE
Fringe around edge of image areas

### ISSUE THREAD

**Issue by ghost** [johannesnoordanus/image2gcode#1]
Hey, Johannes. 

I couldn't find another way to contact you, hence filing an "issue". I've been playing round with several of your projects (grblhud, svg2gcode and image2gcode) and I'm finding them very useful. 

However, with image2gcode I'm finding that I'm getting a "jpeg-like" fringe around the edge of image areas even when I use a 1 bit input image. I'm guessing it's probably my settings or something.

Happy to lend a hand with coding/testing/docu if there's anything simple I can do to help (can code "plain" Python, but not experienced with NumPy).

Thanks for your great work.
-Andy

**Comment by johannesnoordanus** [comment]
Hi Andy,
Thanks for your compliments, alway nice to hear.

I havent seen the effect (artefact) you mention. Could you give an example (image and image2gcode options) so I can look at it?

Regards,
Johannes

**Comment by ghost** [comment]
Hi Johannes, yes I will do a test burn for you and record all the settings,
and take some pics.

Thanks for looking at this.

Regards,
Andy

**Comment by ghost** [comment]
Hey, Johannes!
I've done some testing this weekend, and I have found that there is NO BUG
- it's just a combination of image resolution, burn settings and focus that
was causing the effect. I'm now managing to produce some excellent output
using your tools.

I have another question, if I may regarding svg2gcode - what in the SVG
determines the cut order of paths in gcode?

Thanks,
Andy

**Comment by johannesnoordanus** [comment]
Hi Andy,
It works, that is good news!
(I thought maybe you were using .jpg image files as source, which probably contain the artefacts you mention, and because it is a pixel perfect conversion, the pixels will show in the end result.)

The cut (or engraving) order: I'll have to look that up (I will do so shortly) but it is the tree traversel of the SVG tree, so I expect what comes first within the .svg file. But, I think some ordening occurs in some situations. Also, properties can be added within the SVG tree to set the burn intensity and other specifics. This can be added via svg editors like Inkscape for example, but also directly within the svg file.
Look at the README.md file for examples.
I think you would want to set the path order? I'll look at the implementation for that.

**Comment by ghost** [comment]
Hi Johannes, thanks for your reply! I've actually worked that out too with
some experimentation: it's simply the reverse z-order, so the paths on the
bottom of the stack in inkscape's layers dialog get cut first. I guess that
might change in some circumstances but it's working for me ATM.
Regards, Andy

**Comment by johannesnoordanus** [comment]
Ok, I did not know that, nice you use Inkscape.
The paths are approximated by a set of lines and so converted to gcode movements (to the endpoints of the lines).
So from memory I don't think the order of path rendering is changed by svg2gcode.
But I'll keep a specified path order on the wish list, but will not look into it now.
I am currently busy CNCing parts for a table I designed and am implementing.
Regards,
Johannes

**Comment by ghost** [comment]
Nice one. I'm laser cutting illuminated signs and trinkets. Using inkscape,
svg2gcode and grblhud, so many thanks for those - so much more repeatable
than GUI based tools. Good luck with your table. Are you using a maslow CNC
or a bed-flinger?

**Comment by johannesnoordanus** [comment]
> Nice one. I'm laser cutting illuminated signs and trinkets. Using inkscape, svg2gcode and grblhud, so many thanks for those - so much more repeatable than GUI based tools. Good luck with your table. Are you using a maslow CNC or a bed-flinger?

I'm using a hobby CNC (3018) machine but upgraded it somewhat and will do additional updates soon. 
I'm CNCing oak and it works fine. So I cannot CNC large parts, but I managed to CNC a 30x12x2.5cm piece.
Biggest difficulty was managing FreeCAD (indeed not a command line tool), but I finally got that to work (version 0.21.2).

_"I'm laser cutting illuminated signs and  trinkets"_ That sounds rather intriguing!
_Are you using a maslow CNCor a bed-flinger?"_ The latter.
BTW I updated my profile which includes my email.
